mozfun/hist/extract/udf.sql

```diff
@@ -0,0 +1,174 @@
+CREATE OR REPLACE FUNCTION hist.extract(input STRING) AS (
+  CASE
+  WHEN
+    STARTS_WITH(TRIM(input), '{')
+  THEN
+    -- Input is a histogram in the classic JSON representation.
+    STRUCT(
+      CAST(JSON_EXTRACT_SCALAR(input, '$.bucket_count') AS INT64) AS bucket_count,
+      CAST(JSON_EXTRACT_SCALAR(input, '$.histogram_type') AS INT64) AS histogram_type,
+      CAST(JSON_EXTRACT_SCALAR(input, '$.sum') AS INT64) AS `sum`,
+      ARRAY(
+        SELECT
+          CAST(bound AS INT64)
+        FROM
+          UNNEST(JSON_EXTRACT_ARRAY(input, '$.range')) AS bound
+      ) AS `range`,
+      mozfun.json.extract_int_map(JSON_EXTRACT(input, '$.values')) AS `values`
+    )
+  WHEN
+    ARRAY_LENGTH(SPLIT(input, ';')) = 5
+  THEN
+    -- Input is a compactly encoded boolean histogram like "3;2;5;1,2;0:0,1:5,2:0"
+    STRUCT(
+      CAST(SPLIT(input, ';')[OFFSET(0)] AS INT64) AS bucket_count,
+      CAST(SPLIT(input, ';')[OFFSET(1)] AS INT64) AS histogram_type,
+      CAST(SPLIT(input, ';')[OFFSET(2)] AS INT64) AS `sum`,
+      ARRAY(
+        SELECT
+          CAST(bound AS INT64)
+        FROM
+          UNNEST(SPLIT(SPLIT(input, ';')[OFFSET(3)], ',')) AS bound
+      ) AS `range`,
+      ARRAY(
+        SELECT
+          STRUCT(
+            CAST(SPLIT(entry, ':')[OFFSET(0)] AS INT64) AS key,
+            CAST(SPLIT(entry, ':')[OFFSET(1)] AS INT64) AS value
+          )
+        FROM
+          UNNEST(SPLIT(SPLIT(input, ';')[OFFSET(4)], ',')) AS entry
+      ) AS `values`
+    )
+  WHEN
+    ARRAY_LENGTH(SPLIT(input, ',')) = 2
+  THEN
+    -- Input is a compactly encoded boolean histogram like "0,5"
+    STRUCT(
+      3 AS bucket_count,
+      2 AS histogram_type,
+      CAST(SPLIT(input, ',')[OFFSET(1)] AS INT64) AS `sum`,
+      [1, 2] AS `range`,
+      [
+        STRUCT(0 AS key, CAST(SPLIT(input, ',')[OFFSET(0)] AS INT64) AS value),
+        STRUCT(1 AS key, CAST(SPLIT(input, ',')[OFFSET(1)] AS INT64) AS value),
+        STRUCT(2 AS key, 0 AS value)
+      ] AS `values`
+    )
+  WHEN
+    ARRAY_LENGTH(SPLIT(input, ',')) = 1
+  THEN
+    -- Input is a compactly encoded count histogram like "5"
+    STRUCT(
+      3 AS bucket_count,
+      4 AS histogram_type,
+      CAST(SPLIT(input, ',')[OFFSET(0)] AS INT64) AS `sum`,
+      [1, 2] AS `range`,
+      [
+        STRUCT(0 AS key, CAST(SPLIT(input, ',')[OFFSET(0)] AS INT64) AS value) STRUCT(
+          1 AS key,
+          0 AS INT64
+        ) AS value
+      )
+    ] AS `values`
```

Comment:
```suggestion
      [
        STRUCT(0 AS key, CAST(SPLIT(input, ',')[OFFSET(0)] AS INT64) AS value),
        STRUCT(
          1 AS key,
          0 AS value
        )
    ] AS `values`
```

Might need re-formatting.